### ISSUE TITLE
SpaceAroundOperator: false positive on unary minus (Elixir 1.6.0)

### ISSUE THREAD

**Issue by randycoulman** [rrrene/credo#495]
### Precheck

* For proposing a new feature, be sure to name your issue "Proposal: ..." (n/a)
* For bugs, please do a quick search and make sure the bug has not yet been reported here (done)

### Environment

* Credo version (`mix credo -v`): 0.8.10
* Erlang/Elixir version (`elixir -v`): 1.6.0
* Operating system: macOS Sierra (10.12.6)

### What were you trying to do?

Attempting to upgrade an application from Elixir 1.5.3/credo 0.8.6 -> Elixir 1.6.0.  In Elixir 1.6.0, credo 0.8.6 failed in `Code.toTokens`.  After reviewing the CHANGELOG, I realized that I'd need to also upgrade credo to 0.8.10, so I did that as well.

### Expected outcome

No credo warnings.

### Actual outcome

Running credo on the app resulted in a new SpaceAroundOperator warning that wasn't present previously.  The line in question is:

```
defp parabolic(kwh), do: -abs(kwh) * kwh / (50 * 50 * 2) + 0.5
```

The error message points at the `*` in the `50 * 50` part of the expression, but by changing the code to this:

```
defp parabolic(kwh), do: -1 * abs(kwh) * kwh / (50 * 50 * 2) + 0.5
```

the error goes away.

It looks like the `-` is not being properly interpreted as a unary minus in this case.  There is code to detect in when the following characters are numeric (which is why the `-1` works in the new code).

I also tested with Elixir 1.5.3/credo 0.8.10 and did not see the same warning, so this appears to be a new issue with Elixir 1.6.0.

**Comment by halfdan** [comment]
I'm also seeing warnings with Elixir 1.6.0 when compiling verk (which depends on latest credo): 

```
==> credo
Compiling 161 files (.ex)
warning: this clause cannot match because a previous clause at line 165 always matches
  lib/credo/cli/output/summary.ex:165

warning: module attribute @lint was set but never used
  lib/credo/check/find_lint_attributes.ex:23
```

**Comment by randycoulman** [comment]
@halfdan I see those too, but I think they're unrelated to this issue.  Might be worth opening a separate issue for that so that this thread doesn't get confusing for people.

**Comment by halfdan** [comment]
@randycoulman Ah, fair enough 👍  - only glimpsed over your issue and didn't realise it was describing a different issue.

**Comment by AntonShevel** [comment]
Experienced the same issue with Elixir 1.6.1 and Credo 0.8.10:
```
There are spaces around operators most of the time, but not here.

datetime_with_offset(50 * 60)
```

Here are workaround which are possible
```
datetime_with_offset(-1 * 50 * 60)
datetime_with_offset(-50*60)
```

**Comment by rrrene** [comment]
@randycoulman @halfdan @AntonShevel Could you test with `0.9.0-rc3`? There's still issues to iron out, but I think we're close to release.

**Comment by randycoulman** [comment]
@rrrene Seems to be fixed for the example I originally reported.  Thanks!

**Comment by Cleop** [comment]
@rrrene I'm experiencing this error on `0.9.0-rc3`, with `Elixir 1.6.0`.

I tried setting the Credo dep to 

```
{:credo, github: "rrene/credo"}
```

Based on your comment in https://github.com/rrrene/credo/issues/496#issuecomment-362642775 and now it runs successfully locally doing `mix credo` but still breaks when done as part of [my pre-commit](https://github.com/dwyl/elixir-pre-commit).

**Pass:**
![image](https://user-images.githubusercontent.com/16775804/36319991-8d006574-133c-11e8-8b5a-5b01a04fe1a0.png)

**Fail with no further explanation:**
![image](https://user-images.githubusercontent.com/16775804/36320020-a5c4a23c-133c-11e8-839a-29b2ec10238a.png)

Any advice?

**Comment by rrrene** [comment]
@Cleop There is no evidence of Credo breaking in the screens. Maybe there's a misunderstanding here?

Since Credo is reporting issues on the code, it will exit with an exit status other than zero. This causes "Pre-commit failed on `mix credo`."

You can mute the exit status via `mix credo --mute-exit-status`, but I am unsure if that is useful in your scenario.

**Comment by Cleop** [comment]
@rrrene thanks for getting back to me. So credo was failing with the `SpaceAroundOperators` error but then I applied your recommendation of changing my dep to: `{:credo, github: "rrene/credo"}`.

Now it passes when run simply as `mix credo` but fails when run as part of the pre-commit. Given that all my other pre-commits run fine and that before credo was also not working when run on its own, I presumed that the issue was still with credo. I've found it difficult to debug because when I run the pre-commit (the second screen shot) I'm getting nothing more than `pre-commit failed on mix credo` to help me work out what is going wrong. So perhaps you can tell me whether you think this message is an issue with the pre-commit hook or credo?

**Comment by Cleop** [comment]
Thanks @rrrene I understand it now 👍

**Comment by rrrene** [comment]
@randycoulman @halfdan @AntonShevel @Cleop Again, thanks for reporting this 😀  It should now be fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by rrrene** [comment]
It should be be fixed in the latest version (`v0.9.1`) of Credo.

If it is not, please feel free to re-open this issue!

**Comment by lasseebert** [comment]
The issue I described [above](https://github.com/rrrene/credo/issues/495#issuecomment-378505741) is gone now.

Thanks! :)

**Comment by dabaer** [comment]
Hello, I know this is quite an old issue, but this is flagging in newest elixir and credo. Interestingly I have unary 1 in other parts of that file it doesn't seem to be bothered by.

```
System:
  Credo: 1.7.0-dev-ref.master.b5b51303
  Elixir: 1.13.4
  Erlang: 24.3.4
```

![image](https://user-images.githubusercontent.com/5544713/169720145-8ace3eea-fc29-47bc-a656-2ed7e6021142.png)
